jsapp/js/components/locking/lockingUtils.es6

```diff
@@ -0,0 +1,297 @@
+    typeof foundRow[LOCKING_PROFILE_PROP_NAME] === 'string' &&
+    foundRow[LOCKING_PROFILE_PROP_NAME].length >= 1
+  ) {
+    return getLockingProfile(assetContent, foundRow[LOCKING_PROFILE_PROP_NAME]);
+  }
+
+  return found;
+}
+
+/**
+ * Find locking profile of asset - you don't need to know if asset has
+ * a profile or what's it's name.
+ *
+ * @param {object} assetContent asset's object content property
+ * @returns {object|null} null for no found
+ */
+export function getAssetLockingProfile(assetContent) {
+  let found = null;
+
+  if (
+    assetContent.settings &&
+    typeof assetContent.settings[LOCKING_PROFILE_PROP_NAME] === 'string' &&
+    assetContent.settings[LOCKING_PROFILE_PROP_NAME].length >= 1
+  ) {
+    return getLockingProfile(assetContent, assetContent.settings[LOCKING_PROFILE_PROP_NAME]);
+  }
+
+  return found;
+}
+
+/**
+ * Checks if row has any locking applied, i.e. row has `locking_profile` set and
+ * this locking profile has a definition in the asset content settings. As it is
+ * actually possible for row to have a locking profile name that the asset
+ * doesn't have a definition for.
+ *
+ * If asset has `lock_all` this will also be true.
+ *
+ * @param {object} assetContent asset's object content property
+ * @param {string} rowName - row or group name
+ * @returns {boolean}
+ */
+export function isRowLocked(assetContent, rowName) {
+  // case 1
+  // asset has lock_all
+  if (isAssetAllLocked(assetContent)) {
+    return true;
+  }
+
+  // case 2
+  // row has locking profile that is defined in asset
+  return Boolean(getRowLockingProfile(assetContent, rowName));
+}
+
+/**
+ * Checks if asset is locked, i.e. asset has `lock_all` or `locking_profile`.
+ *
+ * @param {object} assetContent asset's object content property
+ * @returns {boolean} whether form has locking
+ */
+export function isAssetLocked(assetContent) {
+  // case 1
+  // asset has lock_all
+  if (isAssetAllLocked(assetContent)) {
+    return true;
+  }
+
+  // case 2
+  // asset has locking profile and the profile definition exist
+  return Boolean(getAssetLockingProfile(assetContent));
+}
+
+/**
+ * Checks if anything in the asset is locked, i.e. asset has `lock_all` or
+ * `locking_profile` is being set for it or any row.
+ * @param {object} assetContent asset's object content property
+ * @returns {boolean} whether form or any row has locking on it
+ */
+export function hasAssetAnyLocking(assetContent) {
+  // case 1
+  // asset is locked
+  if (isAssetLocked(assetContent)) {
+    return true;
+  }
+
+  // case 2
+  // at least one row has locking profile that is defined in asset
+  const foundRow = assetContent?.survey.find((row) => {
+    return isRowLocked(assetContent, getRowName(row));
+  });
+  return Boolean(foundRow);
+}
+
+/**
+ * Checks if asset has `lock_all`, i.e. everything locked
+ *
+ * @param {object} assetContent asset's object content property
+ * @returns {boolean}
+ */
+export function isAssetAllLocked(assetContent) {
```

Comment:
This would not break anything, just the labels in few places will not say "fully locked".

Reply:
But checking it will not be easy. `lock_all` in the code means that form, all questions and all groups have restrictions from the list `LOCK_ALL_RESTRICTION_NAMES` (currently all known restrictions)

Reply:
I've added a comment #fixed :P